src/custom/components/OrderProgressBar/ContentByExecutionState.tsx

```diff
@@ -37,10 +42,26 @@ interface ExecutionStateProps extends OrderProgressBarProps {
   isSmartContractWallet: boolean
 }
 
+function amountsForMarketPriceFormatted(order: Order) {
+  if (!order?.isUnfillable || !order.amountByCurrentPrice) return {}
+
+  const { currentPrice, orderPrice, percentageDifference } = orderPriceAndCurrentPriceDiff(
+    order,
+    order.amountByCurrentPrice
```

Comment:
If `amountByCurrentPrice` is part of the order, I don't think this should be an extra paramenter.
Only the order should suffice.
Or pass in only the required params rather than the whole order obj

Reply:
I have set the second optional parameter by doing an [isCurrentMarketPriceAmountDefined](https://github.com/cowprotocol/cowswap/blob/153-display-unfillable-price-out-range/src/custom/state/orders/utils.ts#L220) check, since I need `priceAmount` in the first call when the `order` does not yet have the **current market price** set.